### PR TITLE
Skip the very first cold iteration from gpu time measurement 

### PR DESCRIPTION
**Pull request by TomokoKurotobi**
I am new to contribution to the public repository.
I like the new feature to optionally skip running hot iteration for potentially slow solutions that @jichangjichang made.

Since the very first kernel launch may have a big overhead (I have experienced this), how about adding one additional cold iteration for this feature, and start time measurement from 2nd cold iteration?

### CODE REVIEW

File: clients/include/testing_matmul.hpp [hunk 1]
```diff
@@ -2974,11 +2974,14 @@ void testing_matmul_with_bias(const Arguments& arg,
...
                                                   *dWorkspace));
-                    if(arg.skip_slow_solution_ratio)
-                        pre_gpu_time(
-                            arg.use_gpu_timer, event_gpu_time_start, gpu_time_used, stream);
-                    for(int i = 0; i < number_cold_calls; i++)
+                                                  
+                    const int cold_itr_addition = arg.skip_slow_solution_ratio ? 1 : 0;
+                    for(int i = 0; i < number_cold_calls + cold_itr_addition; i++)
                     {
+                        if (i == 1 && arg.skip_slow_solution_ratio){
```
Comment (nakajee) [inline clients/include/testing_matmul.hpp:2981]: pre_gpu_time will not be called if number_cold_calls is 0.
First of all, number_cold_calls needs to be >=1 if arg.skip_slow_solution_ratio is true.
I suggest to set number_cold_calls = 1 if arg.cold_iters == 0 and arg.skip_slow_solution_ratio == true.

**Comment by jichangjichang** [comment]
> I am new to contribution to the public repository. I like the new feature to optionally skip running hot iteration for potentially slow solutions that @jichangjichang made.
> 
> Since the very first kernel launch may have a big overhead (I have experienced this), how about adding one additional cold iteration for this feature, and start time measurement from 2nd cold iteration?

It is too complicated for using. I don't like this change. We won't use just 1 iter for warm-up, and the skip mechanism also can be adjusted tolerance by user. I don't think you need to care the first iter of cold iter.

File: clients/include/testing_matmul.hpp (hunk 1 above)
Comment (TomokoKurotobi) [reply clients/include/testing_matmul.hpp:2981]: number_cold_calls will be at least 1 according to source code, and if skip_slow_solution_ratio is true, I add another addition before for loop, so cold iteration loop will run at least twice.

**Comment by TomokoKurotobi** [comment]
> > I am new to contribution to the public repository. I like the new feature to optionally skip running hot iteration for potentially slow solutions that @jichangjichang made.
> > Since the very first kernel launch may have a big overhead (I have experienced this), how about adding one additional cold iteration for this feature, and start time measurement from 2nd cold iteration?
> 
> It is too complicated for using. I don't like this change. We won't use just 1 iter for warm-up, and the skip mechanism also can be adjusted tolerance by user. I don't think you need to care the first iter of cold iter.

Thanks for your response.
I will close this PR, then

File: clients/include/testing_matmul.hpp (hunk 1 above)
Comment (TomokoKurotobi) [reply clients/include/testing_matmul.hpp:2981]: @nakajee Sorry, I was wrong. Depending on the argument, number_cold_calls can be zero or negative, I think.
image

**Pull request closed without merging**